hazelcast/src/main/java/com/hazelcast/internal/management/ManagementCenterService.java

```diff
@@ -16,213 +16,67 @@
-import com.hazelcast.internal.management.request.WanCheckConsistencyRequest;
-import com.hazelcast.internal.nio.IOUtil;
-import com.hazelcast.internal.util.Clock;
-import com.hazelcast.internal.util.ExceptionUtil;
 import com.hazelcast.internal.util.executor.ExecutorType;
 import com.hazelcast.logging.ILogger;
-import com.hazelcast.map.impl.MapService;
 import com.hazelcast.spi.exception.RetryableException;
 import com.hazelcast.spi.impl.executionservice.ExecutionService;
-import com.hazelcast.spi.impl.operationservice.Operation;
-import com.hazelcast.spi.impl.operationservice.OperationService;
-import com.hazelcast.spi.impl.operationservice.impl.InvocationFuture;
 import com.hazelcast.spi.properties.ClusterProperty;
 import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
 
 import javax.annotation.Nonnull;
-import java.io.IOException;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.io.OutputStream;
-import java.io.OutputStreamWriter;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
-import java.net.URL;
-import java.net.URLConnection;
 import java.util.ArrayList;
-import java.util.Collection;
-import java.util.HashMap;
 import java.util.List;
-import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.BlockingQueue;
-import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
 import java.util.concurrent.LinkedBlockingQueue;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 
 import static com.hazelcast.instance.impl.OutOfMemoryErrorDispatcher.inspectOutOfMemoryError;
-import static com.hazelcast.internal.nio.IOUtil.closeResource;
-import static com.hazelcast.internal.util.EmptyStatement.ignore;
-import static com.hazelcast.internal.util.ExceptionUtil.rethrow;
-import static com.hazelcast.internal.util.JsonUtil.getInt;
-import static com.hazelcast.internal.util.JsonUtil.getObject;
-import static com.hazelcast.internal.util.ThreadUtil.createThreadName;
-import static com.hazelcast.spi.impl.executionservice.ExecutionService.ASYNC_EXECUTOR;
-import static java.net.URLEncoder.encode;
-import static java.nio.charset.StandardCharsets.UTF_8;
 
 /**
  * ManagementCenterService is responsible for sending statistics data to the Management Center.
  */
 public class ManagementCenterService {
+
     public static final String SERVICE_NAME = "hz:core:managementCenterService";
 
-    private static final int HTTP_SUCCESS = 200;
-    private static final int HTTP_NOT_MODIFIED = 304;
-    private static final int CONNECTION_TIMEOUT_MILLIS = 5000;
-    private static final long SLEEP_BETWEEN_POLL_MILLIS = 1000;
-    private static final long DEFAULT_UPDATE_INTERVAL = 3000;
-    private static final long EVENT_SEND_INTERVAL_MILLIS = 1000;
     private static final int EVENT_QUEUE_CAPACITY = 1000;
```

Comment:
If more than 1000 events are logged within ~1 second, we'll loose events. So, probably we need to take the number of the owned partitions into account. Going to address it.

Reply:
It is 1k since introduced in 3.11 (if I'm not mistaken), but I'm not aware of any issue caused by this. I think it is possible to reach this limit for example in a G/B deployment scenario where the DCs are close to each other and we go with >1k partition per member (we send `WanSyncUpdateEvent` per owned partition, the rest of the events doesn't scale). The tricky part here is that the number of partitions owned can change (splits, outages, rolling upgrade etc). If we poll this queue every second, I think `max(1000, totalPartitionCount)` or even `max(1000, totalPartitionCount/3)` can be a good runtime constant for now. If there are plans for extending it beyond WAN, probably we need a more robust and configurable solution for event handling anyways.

Reply:
> If we poll this queue every second, I think max(1000, totalPartitionCount) or even max(1000, totalPartitionCount/3) can be a good runtime constant for now.

Sounds reasonable. Going to implement that logic.

> If there are plans for extending it beyond WAN, probably we need a more robust and configurable solution for event handling anyways.

I'm not aware of such plans, but it's very likely that we may add other event types in future. In this case, we'll need to revisit queue size calculation.

Reply:
Implemented as `max(1000, totalPartitionCount)`.